src/context/SyntheticsStateContext/selectors/tradeboxSelectors/selectTradeboxRelatedMarketsStats.ts

```diff
@@ -0,0 +1,110 @@
+import { BigNumber } from "ethers";
+
+import { selectMarketsInfoData, selectTokensData } from "context/SyntheticsStateContext/selectors/globalSelectors";
+import {
+  getMarketIncreasePositionAmounts,
+  selectTradeboxIncreasePositionAmounts,
+  selectTradeboxToTokenAddress,
+  selectTradeboxTradeFlags,
+} from "context/SyntheticsStateContext/selectors/tradeboxSelectors";
+import { createSelector } from "context/SyntheticsStateContext/utils";
+import { getCappedPositionImpactUsd, getFeeItem } from "domain/synthetics/fees/utils";
+import { getAvailableUsdLiquidityForPosition, isMarketIndexToken } from "domain/synthetics/markets";
+import { getAcceptablePriceByPriceImpact, getMarkPrice } from "domain/synthetics/trade/utils/prices";
+import { EMPTY_ARRAY, getByKey } from "lib/objects";
+import {
+  MarketStat,
+  marketsInfoData2IndexTokenStatsMap,
+} from "domain/synthetics/stats/marketsInfoDataToIndexTokensStats";
+import { values, keyBy, fromPairs } from "lodash";
+
+export type RelatedMarketsStats = {
+  relatedMarketsPositionStats: {
+    [marketTokenAddress: string]: {
+      isEnoughLiquidity: boolean;
+      liquidity: BigNumber;
+      openFees: BigNumber | undefined;
+    };
+  };
+  relatedMarketStats: MarketStat[];
+};
+
+export const selectTradeboxRelatedMarketsStats = createSelector((q) => {
+  const flags = q(selectTradeboxTradeFlags);
+  const indexTokenAddress = q(selectTradeboxToTokenAddress);
+  const tokensData = q(selectTokensData);
+  const marketsInfoData = q(selectMarketsInfoData);
+  const increaseAmounts = q(selectTradeboxIncreasePositionAmounts);
+  const increaseSizeUsd = increaseAmounts?.sizeDeltaUsd;
+
+  const indexToken = getByKey(tokensData, indexTokenAddress);
+
+  const { isPosition, isLong } = flags;
+
+  if (!isPosition || !indexToken || isLong === undefined) {
+    return {
+      relatedMarketsPositionStats: {},
+      relatedMarketStats: [],
+    } as RelatedMarketsStats;
+  }
+
+  const allMarkets = Object.values(marketsInfoData || {}).filter((market) => !market.isSpotOnly && !market.isDisabled);
+
+  const relatedMarkets = allMarkets.filter((market) => isMarketIndexToken(market, indexToken.address));
```

Comment:
I think we can make it separate selector(s) and reuse here https://github.com/gmx-io/gmx-interface/blob/3e48d14f4ba43823e7214f5905feaabe1935cffa/src/context/SyntheticsStateContext/selectors/tradeboxSelectors/selectTradeboxAvailableMarketOptions.ts#L98-L100

Reply:
extracted into a 

```ts
export const selectTradeboxAvailableMarkets = createSelector((q) => {
  const marketsInfoData = q(selectMarketsInfoData);
  const indexTokenAddress = q(selectTradeboxToTokenAddress);

  if (!marketsInfoData || !indexTokenAddress) {
    return EMPTY_ARRAY;
  }

  const allMarkets = values(marketsInfoData).filter((market) => !market.isSpotOnly && !market.isDisabled);

  const availableMarkets = allMarkets.filter((market) => isMarketIndexToken(market, indexTokenAddress));

  return availableMarkets;
});

```